LamestWebserver/LamestWebserver/Caching/ResponseCache.cs

```diff
@@ -0,0 +1,459 @@
+                        return;
+
+                    var response = StringResponses[SortedTimeLeftTillRefresh[j].kvpair.Key];
+
+                    RemoveStringResponseEntry(SortedTimeLeftTillRefresh[j].kvpair.Key, response);
+                }
+
+                // Remove by count and last accesstime regardless of percentiles.
+                j = 0;
+                
+                while (CurrentStringResponseCacheSize + requestedSpace + CacheMakeRoom_AdditionalFreeSpace > MaximumStringResponseCacheSize && StringResponses.Any())
+                {
+                    var response = StringResponses[DateSorted[j].Key];
+
+                    if (response)
+                        RemoveStringResponseEntry(DateSorted[j].Key, response);
+
+                    if (!(CurrentStringResponseCacheSize + requestedSpace + CacheMakeRoom_AdditionalFreeSpace > MaximumStringResponseCacheSize && StringResponses.Any()))
+                        return;
+
+                    response = StringResponses[CountSorted[j].Key];
+
+                    if (response)
+                        RemoveStringResponseEntry(CountSorted[j].Key, response);
+
+                    j++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Removes an entry from the cache.
+        /// <para/> 
+        /// This method can handle non-existent items.
+        /// </summary>
+        /// <param name="key">The key of the entry.</param>
+        /// <param name="value">The response if already retrieved. If null, the response will be retrieved automatically.</param>
+        protected void RemoveStringResponseEntry(string key, ResponseCacheEntry<string> value)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            if (!value)
+                value = StringResponses[key];
+
+            if (!value)
+                return;
+
+            using (StringResponseLock.LockWrite())
+            {
+                CurrentStringResponseCacheSize -= (ulong)value.Response.Length;
+                StringResponses.Remove(key);
+            }
+        }
+
+        /// <summary>
+        /// Adds an Entry to the cache.
+        /// <para/> 
+        /// This method can handle already existent items.
+        /// </summary>
+        /// <param name="key">The key of the response.</param>
+        /// <param name="value">The response.</param>
+        protected void AddStringResponseEntry(string key, ResponseCacheEntry<string> value)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+
+            using (StringResponseLock.LockWrite())
+            {
+                CurrentStringResponseCacheSize -= (ulong)value.Response.Length;
```

Comment:
dafuq? that's just wrong in every possible sense.
subtract old response (if any), add new response.